cloudsmith/data_source_package_download_test.go

```diff
@@ -0,0 +1,324 @@
+package cloudsmith
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+const (
+	apiBaseURL        = "https://api.cloudsmith.io/v1"
+	uploadBaseURL     = "https://upload.cloudsmith.io"
+	apiPackageBaseURL = "https://api-prd.cloudsmith.io/v1"
+)
+
+func TestAccPackageDownload_data(t *testing.T) {
+	t.Parallel()
+
+	namespace := os.Getenv("CLOUDSMITH_NAMESPACE")
+	apiKey := os.Getenv("CLOUDSMITH_API_KEY")
+	userName := "token"
+	repoName := "terraform-acc-download"
+	fileName := "hello.txt"
+	packageName := "hello"
+	packageVersion := "1.0"
+	destinationPath := "."
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					repoSlug := createRepository(apiKey, namespace, repoName, userName)
+					createHelloWorldFile(fileName, packageVersion)
+					uploadPackageToCloudsmith(apiKey, namespace, repoName, fileName, packageName, packageVersion, userName)
+					slugPerm := getPackageSlugPerm(apiKey, namespace, repoName, fileName, userName)
+					client := &http.Client{}
+					waitForPackageSync(client, apiKey, namespace, repoName, slugPerm, userName)
+
+					defer deleteRepository(apiKey, namespace, repoSlug, userName) // Delete repository after test
```

Comment:
this is going to delete the repository immediately when this function finishes, not when the test finishes, is that what you want?